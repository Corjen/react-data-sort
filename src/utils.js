import sortBy from 'lodash.sortby'

export const calculatePages = (items, itemsPerPage) => {
  if (typeof items !== 'number' || typeof itemsPerPage !== 'number') {
    throw new Error('items and itemsPerPage should both be a number')
  }
  return Math.ceil(items / itemsPerPage)
}

export const sortData = (data, key, direction = 'asc') => {
  const sorted = sortBy(data, key)
  return direction === 'desc' ? sorted.reverse() : sorted
}

export const paginateData = (data, activePage, itemsPerPage) => {
  const from = activePage * itemsPerPage
  const to = from + itemsPerPage
  if (typeof activePage !== 'number' || typeof itemsPerPage !== 'number') {
    throw new Error('activePage and itemsPerPage should both be a number')
  }
  return data.slice(from, to)
}
