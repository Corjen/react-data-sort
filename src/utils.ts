import sortBy from 'lodash.sortby'

export const calculatePages = (items: number, itemsPerPage: number) => {
  if (typeof items !== 'number' || typeof itemsPerPage !== 'number') {
    throw new Error('items and itemsPerPage should both be a number')
  }
  return Math.ceil(items / itemsPerPage)
}

export const sortData = (data: any[], key: string, direction = 'asc') => {
  const sorted = sortBy(data, key)
  return direction === 'desc' ? sorted.reverse() : sorted
}

export const paginateData = (data: any[], activePage: number, itemsPerPage: number) => {
  const from = activePage * itemsPerPage
  const to = from + itemsPerPage
  if (typeof activePage !== 'number' || typeof itemsPerPage !== 'number') {
    throw new Error('activePage and itemsPerPage should both be a number')
  }
  return data.slice(from, to)
}
