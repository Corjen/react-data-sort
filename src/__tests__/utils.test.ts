import { calculatePages, sortData, paginateData } from '../utils'

describe('calculatePages', () => {
  it('should calculate the right amount of pages', () => {
    expect(calculatePages(10, 2)).toEqual(5)
    expect(calculatePages(20, 4)).toEqual(5)
    expect(calculatePages(10, 3)).toEqual(4)
    expect(calculatePages(1, 10)).toEqual(1)
  })
})

describe('sortData', () => {
  const data = [{ id: 2, name: 'b' }, { id: 1, name: 'c' }, { id: 3, name: 'a' }]

  it('should order the data by id', () => {
    expect(sortData(data, 'id')).toEqual([data[1], data[0], data[2]])
    expect(sortData(data, 'id', 'asc')).toEqual([data[1], data[0], data[2]])
  })

  it('should order the data by id in reversed order', () => {
    expect(sortData(data, 'id', 'desc')).toEqual([data[2], data[0], data[1]])
  })

  it('should order the data by name', () => {
    expect(sortData(data, 'name')).toEqual([data[2], data[0], data[1]])
  })
  it('should order the data by name in reversed order', () => {
    expect(sortData(data, 'name', 'desc')).toEqual([data[1], data[0], data[2]])
  })
  it('should return the data if the key does not exist', () => {
    expect(sortData(data, 'key')).toEqual(data)
  })
})

describe('paginateData', () => {
  const data = [{ id: 2 }, { id: 1 }, { id: 3 }]
  it('should return the right slice', () => {
    expect(paginateData(data, 0, 1)).toEqual([data[0]])
    expect(paginateData(data, 2, 1)).toEqual([data[2]])
    expect(paginateData(data, 0, 2)).toEqual([data[0], data[1]])
  })
  it('should return the unsliced data', () => {
    expect(paginateData(data, 0, 1)).toEqual([data[0]])
  })
})
