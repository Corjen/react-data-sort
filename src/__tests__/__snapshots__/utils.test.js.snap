// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculatePages should calculate the right amount of pages 1`] = `"items and itemsPerPage should both be a number"`;

exports[`paginateData should return the unsliced data 1`] = `"activePage and itemsPerPage should both be a number"`;
